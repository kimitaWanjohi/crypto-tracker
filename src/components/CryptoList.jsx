import CryptoCard from "./CryptoCard";
import Search from "./Search";

export default function CryptoList({ cryptos }) {
  return (
    <>
      <div className="container mx-auto px-4 mb-4">
        <div className="w-full my-2 py-2">
          <Search />
        </div>
        <div className="grid grid-cols-2 md:grid-cols-2 lg:grid-cols-3 gap-4">
          {cryptos.map((crypto) => (
            <CryptoCard key={crypto.name} crypto={crypto} />
          ))}
        </div>
      </div>
    </>
  );
}
