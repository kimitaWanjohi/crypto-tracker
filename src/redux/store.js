import { configureStore } from "@reduxjs/toolkit";
import cryptosReducer from "./cryptos/slice";

export default configureStore({
  reducer: {
    cryptos: cryptosReducer,
  },
});
