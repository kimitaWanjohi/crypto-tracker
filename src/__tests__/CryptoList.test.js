import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../redux/store";
import CryptoList from "../components/CryptoList";

const mockCrypto = [
  {
    symbol: "611",
    name: "SixEleven",
    name_full: "SixEleven (611)",
    max_supply: 611000,
    icon_url: "https://assets.coinlayer.com/icons/611.png",
  },
  {
    symbol: "ABC",
    name: "AB-Chain",
    name_full: "AB-Chain (ABC)",
    max_supply: 210000000,
    icon_url: "https://assets.coinlayer.com/icons/ABC.png",
  },
  {
    symbol: "ACP",
    name: "Anarchists Prime",
    name_full: "Anarchists Prime (ACP)",
    max_supply: 53760000,
    icon_url: "https://assets.coinlayer.com/icons/ACP.png",
  },
  {
    symbol: "ACT",
    name: "ACT",
    name_full: "ACT (ACT)",
    max_supply: 10000000000,
    icon_url: "https://assets.coinlayer.com/icons/ACT.png",
  },
  {
    symbol: "ACT*",
    name: "Achain",
    name_full: "Achain (ACT*)",
    max_supply: 1000000000,
    icon_url: "https://assets.coinlayer.com/icons/ACT*.png",
  },
];

describe("CryptoList cryptos={mockCrypto}", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CryptoList cryptos={mockCrypto} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders a list of missions", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CryptoList cryptos={mockCrypto} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
});
