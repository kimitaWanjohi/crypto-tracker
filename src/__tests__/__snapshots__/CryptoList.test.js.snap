// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoList cryptos={mockCrypto} renders correctly 1`] = `
<div
  className="container mx-auto px-4 mb-4"
>
  <div
    className="w-full my-2 py-2"
  >
    <div
      className="w-full py-4 bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 sticky top-40 z-5"
    >
      <input
        className="w-full bg-gray-200 dark:bg-gray-800  text-gray-500 rounded-lg shadow-md p-4"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </div>
  </div>
  <div
    className="grid grid-cols-2 md:grid-cols-2 lg:grid-cols-3 gap-4"
  >
    <div
      className="w-full relative bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 hover:dark:bg-gray-600 "
      onClick={[Function]}
    >
      <div
        className="absolute top-1 right-1 dark:text-gray-400"
      >
        <svg
          className="text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col justify-between items-start"
      >
        <div
          className="flex flex-col justify-between"
        >
          <img
            alt="SixEleven"
            className="w-10 h-10 rounded-full"
            src="https://assets.coinlayer.com/icons/611.png"
          />
          <div
            className="flex flex-col"
          >
            <span
              className="text-lg font-bold"
            >
              SixEleven (611)
            </span>
            <span
              className="text-sm text-gray-400"
            >
              611
            </span>
          </div>
        </div>
        <div
          className="flex flex-col md:flex-row items-center"
        >
          <span
            className="text-lg font-bold mr-2"
          >
            max supply
            <br />
            611,000
          </span>
        </div>
      </div>
    </div>
    <div
      className="w-full relative bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 hover:dark:bg-gray-600 "
      onClick={[Function]}
    >
      <div
        className="absolute top-1 right-1 dark:text-gray-400"
      >
        <svg
          className="text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col justify-between items-start"
      >
        <div
          className="flex flex-col justify-between"
        >
          <img
            alt="AB-Chain"
            className="w-10 h-10 rounded-full"
            src="https://assets.coinlayer.com/icons/ABC.png"
          />
          <div
            className="flex flex-col"
          >
            <span
              className="text-lg font-bold"
            >
              AB-Chain (ABC)
            </span>
            <span
              className="text-sm text-gray-400"
            >
              ABC
            </span>
          </div>
        </div>
        <div
          className="flex flex-col md:flex-row items-center"
        >
          <span
            className="text-lg font-bold mr-2"
          >
            max supply
            <br />
            210,000,000
          </span>
        </div>
      </div>
    </div>
    <div
      className="w-full relative bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 hover:dark:bg-gray-600 "
      onClick={[Function]}
    >
      <div
        className="absolute top-1 right-1 dark:text-gray-400"
      >
        <svg
          className="text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col justify-between items-start"
      >
        <div
          className="flex flex-col justify-between"
        >
          <img
            alt="Anarchists Prime"
            className="w-10 h-10 rounded-full"
            src="https://assets.coinlayer.com/icons/ACP.png"
          />
          <div
            className="flex flex-col"
          >
            <span
              className="text-lg font-bold"
            >
              Anarchists Prime (ACP)
            </span>
            <span
              className="text-sm text-gray-400"
            >
              ACP
            </span>
          </div>
        </div>
        <div
          className="flex flex-col md:flex-row items-center"
        >
          <span
            className="text-lg font-bold mr-2"
          >
            max supply
            <br />
            53,760,000
          </span>
        </div>
      </div>
    </div>
    <div
      className="w-full relative bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 hover:dark:bg-gray-600 "
      onClick={[Function]}
    >
      <div
        className="absolute top-1 right-1 dark:text-gray-400"
      >
        <svg
          className="text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col justify-between items-start"
      >
        <div
          className="flex flex-col justify-between"
        >
          <img
            alt="ACT"
            className="w-10 h-10 rounded-full"
            src="https://assets.coinlayer.com/icons/ACT.png"
          />
          <div
            className="flex flex-col"
          >
            <span
              className="text-lg font-bold"
            >
              ACT (ACT)
            </span>
            <span
              className="text-sm text-gray-400"
            >
              ACT
            </span>
          </div>
        </div>
        <div
          className="flex flex-col md:flex-row items-center"
        >
          <span
            className="text-lg font-bold mr-2"
          >
            max supply
            <br />
            10,000,000,000
          </span>
        </div>
      </div>
    </div>
    <div
      className="w-full relative bg-gray-200 dark:bg-gray-700 rounded-lg shadow-md p-4 hover:dark:bg-gray-600 "
      onClick={[Function]}
    >
      <div
        className="absolute top-1 right-1 dark:text-gray-400"
      >
        <svg
          className="text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col justify-between items-start"
      >
        <div
          className="flex flex-col justify-between"
        >
          <img
            alt="Achain"
            className="w-10 h-10 rounded-full"
            src="https://assets.coinlayer.com/icons/ACT*.png"
          />
          <div
            className="flex flex-col"
          >
            <span
              className="text-lg font-bold"
            >
              Achain (ACT*)
            </span>
            <span
              className="text-sm text-gray-400"
            >
              ACT*
            </span>
          </div>
        </div>
        <div
          className="flex flex-col md:flex-row items-center"
        >
          <span
            className="text-lg font-bold mr-2"
          >
            max supply
            <br />
            1,000,000,000
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
